File: tmdb-core/src/main/java/dev/reuss/tmdb/common/image/TvSeasonImages.java
package dev.reuss.tmdb.common.image;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Image metadata for a TMDB TV season.
 *
 * @param id      TMDB TV season id
 * @param posters season poster images
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeasonImages(
        int id,
        List<TmdbImage> posters
) implements TmdbModel {

    public TvSeasonImages {
        posters = TmdbCollections.list(posters);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/image/TvSeasonImages.kt
package dev.reuss.tmdb.common.image

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Image metadata for a TMDB TV season.
 *
 * @property id      TMDB TV season id
 * @property posters season poster images
 */
@JvmRecord
data class TvSeasonImages(
    @all:JsonProperty("id")
    val id: Int,

    @all:JsonProperty("posters")
    val posters: List<TmdbImage>?
) : TmdbModel
